client/client.go

```diff
@@ -1216,12 +1208,14 @@ func (c *Client) syncConsul() {
 					if taskState.State == structs.TaskStateRunning {
 						if tr, ok := ar.tasks[taskName]; ok {
 							for _, service := range tr.task.Services {
-								services[service.ID(ar.alloc.ID, tr.task.Name)] = struct{}{}
+								svcIdentifier := fmt.Sprintf("%s-%s", ar.alloc.ID, tr.task.Name)
```

Comment:
Can we reverse this order so allocations for tasks will lexically sort adjacent to one another? In Vault, we're using the following format for the Consul service: `fmt.Sprintf("%s:%s:%d", c.serviceName, c.advertiseHost, c.advertisePort)`.